features/scv/jceks-impl/src/main/java/org/opennms/features/scv/jceks/JCEKSSecureCredentialsVault.java

```diff
@@ -136,6 +146,7 @@ public void setCredentials(String alias, Credentials credentials) {
             KeyStore.PasswordProtection keyStorePP = new KeyStore.PasswordProtection(m_password);
             m_keystore.setEntry(alias, new KeyStore.SecretKeyEntry(generatedSecret), keyStorePP);
             writeKeystoreToDisk();
+            m_credentialsCache.put(alias, credentials);
```

Comment:
need to do this in thread safe manner.

Reply:
OK. I was wondering if I should use a synchronizedMap or ConcurrentHashMap. Looks like for the size of the matter  using synchronized should be enough.